newbank/server/NewBankClientHandler.java

```diff
@@ -97,14 +97,27 @@ private void handleLogIn() throws IOException{
 			// if the user is authenticated then get requests from the user and process them
 
 			if(customer != null) {
+				// to break out of the enter password loop
+				i=-1;
 				out.println("Log In Successful");
-				while(true) {
+				boolean showMenu2=(true);
+				while(showMenu2) {
 
 					printMenu2(customer);
 					String request = in.readLine();
-					System.out.println("Request from " + customer.getKey());
+					out.println("\nRequest from " + customer.getKey());
 					String responce = bank.processRequest(customer, request);
 					out.println(responce);
+					String account = bank.accountType(request);
+					// when a customer has selected an account to view, show them new options
+					if(responce.contains("Withdraw")){
```

Comment:
This is OK for now, but generally I wouldn't recommend checking like this - it's a bit ropey because if someone changes the message text, or adds a new response type that also contains the word 'withdraw', it will break. Better would be something like returning a HashMap where the key is a unique identifier for that response, and the value is a text message for the user.

Reply:
Yeah I agree. I did it this way so I could get onto including the deposit functionality but perhaps @joa013 or @MK2004-png could take a look at changing this?

Reply:
OK I was slightly confused as to what this was checking for - it's actually just checking that you got into the `showAccountOptions` method I guess, and chooses 'withdraw' as an arbitrary word to check. The main point stands though, I think - I would try to find a more robust way of checking which route you've gone down.